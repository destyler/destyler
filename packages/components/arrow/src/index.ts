export { DestylerArrow, type DestylerArrowProps } from './arrow'
