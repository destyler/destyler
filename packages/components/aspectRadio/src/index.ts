export * from './aspectRadio'
export * from './resolver'
