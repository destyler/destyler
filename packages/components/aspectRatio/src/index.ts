export * from './aspectRatio'
export * from './resolver'
