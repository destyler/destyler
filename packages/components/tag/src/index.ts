export * from './tag'
export * from './resolver'
