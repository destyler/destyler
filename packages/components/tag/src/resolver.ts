import type { ComponentResolver } from 'unplugin-vue-components'
import { name } from '../package.json'

const packageName = name

export function DestylerTagResolver(): ComponentResolver {
  return {
    type: 'component',
    resolve: (name: string) => {
      if (name === 'DestylerTag')
        return { name, from: packageName }
    },
  }
}
