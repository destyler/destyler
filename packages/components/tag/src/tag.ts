import type { PropType } from 'vue'
import { defineComponent, h, ref } from 'vue'
import type { MaybeArray } from '@destyler/shared'
import { call, resolveWrappedSlot } from '@destyler/shared'

const destylerTagProps = {
  'closable': {
    type: Boolean as PropType<boolean>,
  },
  'disabled': {
    type: Boolean as PropType<boolean | undefined>,
    default: undefined,
  },
  'triggerClickOnClose': {
    type: Boolean,
  },
  'checked': {
    type: Boolean as PropType<boolean>,
  },
  'checkable': {
    type: Boolean as PropType<boolean>,
  },
  'onClose': {
    type: [Array, Function] as PropType<MaybeArray<(e: MouseEvent) => void>>,
  },
  'onUpdateChecked': {
    type: Function as PropType<(checked: boolean) => void>,
  },
  'onUpdate:checked': {
    type: Function as PropType<(checked: boolean) => void>,
  },
  'onCheckedChange': {
    type: Function as PropType<(checked: boolean) => void>,
  },
  'onMouseenter': {
    type: Function as PropType<(e: MouseEvent) => void>,
  },
  'onMouseleave': {
    type: Function as PropType<(e: MouseEvent) => void>,
  },
  // private
  'internalCloseFocusable': {
    type: Boolean,
    default: true,
  },
}

const DestylerTag = defineComponent({
  name: 'DestylerTag',
  props: destylerTagProps,
  setup(props) {
    const content = ref<HTMLElement | null>(null)

    function handleClick(e: MouseEvent): void {
      if (!props.disabled) {
        if (props.checkable) {
          const {
            checked,
            onCheckedChange,
            onUpdateChecked,
            'onUpdate:checked': _onUpdateChecked,
          } = props
          if (onUpdateChecked)
            onUpdateChecked(!checked)
          if (_onUpdateChecked)
            _onUpdateChecked(!checked)
          if (onCheckedChange)
            onCheckedChange(!checked)
        }
      }
    }
    function handleCloseClick(e: MouseEvent): void {
      if (!props.triggerClickOnClose)
        e.stopPropagation()

      if (!props.disabled) {
        const { onClose } = props
        if (onClose)
          call(onClose, e)
      }
    }

    return {
      content,
      handleClick,
      handleCloseClick,
    }
  },
  render() {
    const avatarNode = resolveWrappedSlot(
      this.$slots.avatar,
      children =>
        children && (
          h('div', {
            destyler: 'tag-avatar',
          }, children)
        ),
    )
    const iconNode = resolveWrappedSlot(
      this.$slots.icon,
      children =>
        children && (
          h('div', {
            destyler: 'tag-icon',
          }, children)
        ),
    )
    const closeNode = resolveWrappedSlot(
      this.$slots.close,
      children =>
        children && (
          h('div', {
            destyler: 'tag-close',
            onClick: this.handleCloseClick,
            focusable: this.$props.internalCloseFocusable,
          }, children)
        ),
    )
    return h('div', {
      'destyler': 'tag',
      'data-checked': this.checked,
      'onClick': this.handleClick,
      'onMouseenter': this.$props.onMouseenter,
      'onMouseleave': this.$props.onMouseleave,
    }, [
      (iconNode || avatarNode),
      h('span', {
        destyler: 'tag-content',
        ref: 'content',
      }, this.$slots.default?.()),
      !this.$props.checkable && this.$props.closable
        ? closeNode
        : null,
    ])
  },
})

export {
  DestylerTag,
}
