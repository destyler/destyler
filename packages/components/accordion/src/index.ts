import { DestylerAccordionResolver } from './resolver'
import DestylerAccordionRoot from './AccordionRoot.vue'
import DestylerAccordionItem from './AccordionItem.vue'
import { selected, useAccordionSelected } from './composition'
import destylerAccordionPresets from './presets'

export * from './props'

export {
  selected,
  DestylerAccordionRoot,
  DestylerAccordionItem,
  DestylerAccordionResolver,
  useAccordionSelected,
  destylerAccordionPresets,
}
