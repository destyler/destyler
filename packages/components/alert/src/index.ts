import { DestylerAccordionResolver } from './resolver'
import DestylerAlert from './Alert'

export {
  DestylerAlert,
  DestylerAccordionResolver,
}
