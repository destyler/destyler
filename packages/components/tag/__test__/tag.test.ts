import { describe, expect, it } from 'vitest'
import { mount } from '@vue/test-utils'
import { DestylerTag } from '../src'

describe('basic', () => {
  it('should work with import on demand', () => {
    mount(DestylerTag)
  })

  it('should have a role of "alert"', async () => {
    const wrapper = mount(DestylerTag)
    expect(wrapper.find('div').attributes('role')).toBe('tag')
    wrapper.unmount()
  })

  it('shouldn\'t closed when on-close prop returns false', async () => {
    const wrapper = mount(DestylerTag, {
      props: { closable: true, onClose: () => false },
    })
    const closeBtn = wrapper.find('div')
    await closeBtn.trigger('click')

    expect(wrapper.find('div').exists()).toBe(true)
    wrapper.unmount()
  })
})
