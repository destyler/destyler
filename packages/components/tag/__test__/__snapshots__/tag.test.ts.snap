// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Basic > should work with default slot 1`] = `"<span destyler="tag-content">default</span>"`;

exports[`basic > should work with default slot 1`] = `"<span destyler="tag-content">default</span>"`;
